### PR TITLE
Add mamba and conda-libmamba-solver to Miniforge (Mambaforge is now identical)

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
cc @jaimergp, @jezdez, @wolfv

File: scripts/build.sh
Comment (hmaarrfk) [inline scripts/build.sh:25]: This is probably why the last tag was failing.

**Review by hmaarrfk — APPROVED** [review]

File: scripts/build.sh
Comment (jaimergp) [reply scripts/build.sh:25]: If we merge https://github.com/conda-forge/constructor-feedstock/pull/61 we don't need this pip install override, right?

File: Miniforge3/construct.yaml
Comment (jaimergp) [inline Miniforge3/construct.yaml:32]: This will pull `libmambapy` plus its deps so I wonder if just the presence of `mamba` as a command is enough to justify the existence of Mambaforge. Like, if we merge, Miniforge and Mambaforge are almost the same. IIRC, Mambaforge was provided separately because it shipped more deps.

File: scripts/build.sh
Comment (jaimergp) [inline scripts/build.sh:25]: conda-forge/constructor-feedstock#61 was merged, so we can remove this in a couple hours.

```suggestion
```

**Review by jezdez — COMMENTED** [review]
A few notes on this:

**Some context first:** The new solver integration was released as a separate package as a means for users to "opt-in" into the experimental phase of the engineering effort, basically for us to be able to collect feedback from real world environments. The goal is to gain the confidence that our integration layer with libmamba indeed works as closely as possible to conda's default solver. It's not just a straight port of libmamba but does extra steps to be closer to what users expect from conda. The extra installation step was a design decision to make sure we're not accidentally cargo-cult the experimental solver into "production environments" such as conda-forge.

**The miniconda 4.12.0 installer will not automatically install conda-libmamba-solver.** It largely depends on the real world user feedback, but I'm expecting it to be added in the next release of miniconda at the end of Q2, start of Q3 time frame instead, when we hope to have finalized the experiment.

For conda-forge **I would strongly suggest to wait for at least one or two bugfix releases before adding conda-libmamba-solver to miniforge**, simply because it would make it easier for Jaime, Wolf and me to ship breaking changes in conda-libmamba-solver when they become necessary following user feedback.

Regarding mambaforge and how conda-libmamba-solver fits into this, I want to encourage us to think about this before we make decisions (like merging this PR here) before we regret it afterwards. **mambaforge/miniforge and mamba are reasonable alternatives to miniconda and conda** and I'd rather gain more experience with conda-libmamba-solver before we change the installers that a lot of users use.

The alternative to this PR right now is: `conda install conda-libmamba-solver`.

Could we maybe put this PR on hold for a bit?

**Comment by jezdez** [comment]
I take the merging of #280 as a yes :) Thanks @isuruf!

**Comment by jakirkham** [comment]
IIUC sounds like we are now good to include `conda-libmamba-solver` ( https://github.com/conda-forge/miniforge/issues/284#issuecomment-1582347330 )

Would we like to revitalize this PR or start a new one?

**Comment by jaimergp** [comment]
@jakirkham - I added a comment on 284 to figure out how we want to move forward! Thanks for the reminder!

**Comment by jaimergp** [comment]
Updated PR as discussed in #284. 

Note that I am _not_ copying the installers yet. The name of the installer, as taken by `constructor` will also determine the default install location, so some people might be expecting `Mambaforge` there. 

I suggest we advertise this in the release notes and maybe in other channels (blog post, Twitter, Element, etc), and then change it to a copy in the next release or two.

**Comment by jaimergp** [comment]
This is ready for review @conda-forge/miniforge. Thanks!

**Comment by jakirkham** [comment]
FWIW LGTM. Thanks Jaime! 🙏

**Review by jezdez — APPROVED** [review]

**Comment by jakirkham** [comment]
Friendly nudge @hmaarrfk 😉

**Comment by hmaarrfk** [comment]
Sorry, trying to enable boa compatibility tests on miniforge too.

**Review by hmaarrfk — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks everyone! 🙏

**Comment by hmaarrfk** [comment]
Release is now out. Give it a try!

**Comment by jaimergp** [comment]
Thanks Mark! Looks like it's gone live in macOS and Windows and no one has complained in Element yet :D I have sent a PR to change the base Miniforge install in our Linux Docker images at https://github.com/conda-forge/docker-images/pull/239

I also updated the release description a bit to add a couple extra details. Hope that's ok!

**Comment by jakirkham** [comment]
Noticed the tag is 23.3.1. Am wondering if it should be 23.8.0

File: Miniforge3/construct.yaml
Comment (jakirkham) [inline Miniforge3/construct.yaml:35]: Ah nvm looks like the version is tied to how Conda & friends are versioned. Submitted PR ( https://github.com/conda-forge/miniforge/pull/480 ) to address this

**Pull request merged**
